### ISSUE TITLE
Name filter edit box is unlabeled and extra unlabeled focus is provided for the filter dialog

### ISSUE THREAD

**Issue by gkhadK** [vmware/clarity#4664]
**Clarity Version:** 3.1.1
**WCAG 2.1 Guideline:** 4.1.2: Name, role, value

**Test Environment:**
- Test URL: https://clarity.design/documentation/datagrid/
- OS: Windows 10
- User Agents/Assistive Technologies: Firefox 76.0
- Screen reader: NVDA 2020.1

**Steps to reproduce:**
1. Open the test URL.
2. Navigate to any one of the datagrids with filter options.
3. Press any one of the filter buttons and navigate through the filter dialog.

**Expected result:**
User should be able to navigate through the filter dialog with appropriate announcement while navigating through the content.

**Actual result:**
- Unlabeled edit box is encountered in the filter popup.
- Overall dialog is focused and announced as section. Note: The popup window behaves like a dialog but it&#039;s not announced as dialog to the screen reader user.

**Comment by pratikgala** [comment]
I have noticed the same.

On a voice over reader, the input field is read as "edit text blank."

![Screenshot 2020-06-26 at 17 08 35](https://user-images.githubusercontent.com/3613248/85909988-5b896400-b7d1-11ea-8ea6-6c074683b3e9.png)


The entire div does not have a role name associated with it, which can confuse the user as what the pop-up does.

![Screenshot 2020-06-26 at 16 37 25](https://user-images.githubusercontent.com/3613248/85910015-7cea5000-b7d1-11ea-9898-6460b5a2e7c8.png)
